packages/cli/src/lib/push/addCommon.ts

```diff
@@ -0,0 +1,94 @@
+import {CliUx} from '@oclif/core';
+import {red, dim, green} from 'chalk';
+import {BuiltInTransformers, errors} from '@coveo/push-api-client';
+import {normalizeInvalidFields} from '../flags/sourceCommonFlags';
+import {pluralizeIfNeeded} from '../utils/string';
+import dedent from 'ts-dedent';
+
+const allowedCharRegExp = new RegExp('^[a-z]+[a-z0-9_]*$');
+interface PrintableUnsupportedField {
+  original: string;
+  normalized: string;
+  valid: boolean;
+}
+
+export const formatErrorMessage = (err: unknown) => {
+  const {UnsupportedFieldError} = errors;
+  if (err instanceof UnsupportedFieldError) {
+    const normalizations = getNormalizations(err.unsupportedFields);
+    const fixables = normalizations.flatMap(colorizeFixableField);
+    const unfixables = normalizations.flatMap(colorizeUnFixableField);
```

Comment:
You end up going through the array twice, you're probably due to do a method that follows this signature:

```ts
const {fixables, unfixables} = colorizeFields(normalizations);
```

Reply:
true, but I find it harder to read
![image](https://user-images.githubusercontent.com/12199712/170506127-3ccc4e43-56cc-4cde-be9a-4459b1a5aca9.png)

Reply:
If we had a [partition](https://underscorejs.org/#partition) filter, it would have been cleaner.

Reply:
You just have to remove the unfixable normalize formating in another func:
https://www.typescriptlang.org/play?ssl=17&ssc=110&pln=17&pc=56#code/C4TwDgpgBACgTgSwHbAIYCMA2ECqSDOArmGAPZzAQAmAYghJlVALxSpIgDcAUAMakFgUfpnIIAXhDoMq+FlAAUAM3qN8ALliIUGbHiIlylWqqoBtALoBKFgD4oAb25RhA-EJUAPXRA1bkaFi4BMRkFNTSjJbyljwu-IJQhEhePn7wAT76oUYRptGssc5QSuSKCe4lplCkSlUy+DZOLi4IdcqmAHQAbqiYCFRNxS1V3kH4nWCE+AAWCg5QnUsqMgA0UEjkALZ9EtSaAOZwEBBIHTKdm3A7-ZKDUAC+VnEtD1AM+NDNI0kpCGPYCZTWbzYY-JadFaMdZXG57KiaUrXVDAABy212dxo5DwqSC50Ylwxt2oVjBLieLwpxQexWOwEIcCQjlGaXWyTxgMe3FpfDcHgxaOJ8OxcFx-x88gUyCmwE07m0BxszHsMsIQmYmqgAHJtVAAPxQKgILYKbUKCBbMCgeqMDaoLYQKzamyaY5UaVIWVWIA